presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxStatsDumper.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.jmx;
+
+import com.facebook.presto.spi.SchemaTableName;
+import io.airlift.log.Logger;
+
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class JmxStatsDumper
+        implements Runnable
+{
+    private static final Logger log = Logger.get(JmxStatsDumper.class);
+
+    private final JmxStatsHolder jmxStatsHolder;
+    private final JmxMetadata jmxMetadata;
+    private final JmxRecordSetProvider jmxRecordSetProvider;
+    private final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+    private final long period;
+    private long nextDump = -1;
+
+    public JmxStatsDumper(JmxStatsHolder jmxStatsHolder, JmxMetadata jmxMetadata, JmxRecordSetProvider jmxRecordSetProvider, long period)
+    {
+        this.jmxStatsHolder = requireNonNull(jmxStatsHolder, "jmxStatsHolder is null");
+        this.jmxMetadata = requireNonNull(jmxMetadata, "jmxMetadata is null");
+        this.jmxRecordSetProvider = requireNonNull(jmxRecordSetProvider, "jmxRecordSetProvider is null");
+        this.period = requireNonNull(period, "period is null");
+    }
+
+    public void start()
+    {
+        long nowTs = System.currentTimeMillis();
+        nextDump = roundToPeriod(nowTs) + period;
+        schedule(nowTs);
+    }
+
+    private void schedule(long nowTs)
+    {
+        executor.schedule(this, Math.max(nextDump - nowTs, 0), TimeUnit.MILLISECONDS);
```

Comment:
maybe just use `ScheduledExecutorService.scheduleAtFixedRate` instead of doing this manually.